File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        //prefix
        vector<int>pref(n);
        pref[0]=nums[0];
        for(int i=1;i<n;i++){
            pref[i]=pref[i-1]*nums[i];
        }
        //suffix
        vector<int> suff(n);
        suff[n-1]=nums[n-1];
        for(int i = n-2;i>=0;i--){
            suff[i] = suff[i+1]*nums[i];
        }
        vector<int> output(n);
        for(int i = 0;i<n;i++){
            int left = (i == 0) ? 1 : pref[i-1];
            int right = (i == n - 1) ? 1 : suff[i+1];
            output[i] = left * right;
        }
        return output;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] height) {
        int len = height.length;
        int output = 0;
        int left = 0, right = height.length-1;
        while(left<right){
            output = Math.max(output, (right-left)*Math.min(height[left], height[right]));
            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return output;
    }
}